yubihsm/objects.py

```diff
@@ -733,9 +733,9 @@ def sign_ssh_certificate(
 
         :param template_id: The ID of the SSH TEMPLATE to use.
         :param request: The SSH certificate request.
-        :return: The signed SSH certificate.
+        :return: The SSH certificate signature.
         """
-        msg = struct.pack("!HH", self.id, template_id) + request
+        msg = struct.pack("!HHB", self.id, template_id, algorithm.value) + request
```

Comment:
Please change `algorithm.value` to just `algorithm` here, the rest isn't needed.